src/MakoIoT.Device/IoTDevice.cs

```diff
@@ -21,15 +21,12 @@ public void Start()
             if (IsRegistered(ServiceProvider, typeof(IDeviceStartBehavior)))
             {
                 var behaviors = ServiceProvider.GetServices(typeof(IDeviceStartBehavior));
-                if (behaviors.Length > 1)
+                foreach (IDeviceStartBehavior behavior in behaviors)
                 {
-                    throw new InvalidOperationException($"More than one {nameof(IDeviceStartBehavior)} is registered");
-                }
-
-                var behavior = (IDeviceStartBehavior) behaviors[0];
-                if (!behavior.DeviceStarting())
-                {
-                    return;
+                    if (!behavior.DeviceStarting())
+                    {
+                        return;
```

Comment:
Just wondering, return or continue?

Reply:
The original implementation returned on false. I assumed this was by design to signal a failure.

In my personal use case for this I will have three `IDeviceStartBehavior`

- Common startup with core requirements (display, gpio, etc.)
- Network startup (to connect to wifi, enable AP mode and reboot if not configured, etc.)
- Application startup (depends on success from the previous behaviors)

So returning on failure makes sense to me since they all need to execute successfully and in order.

That said I could add a logging statement so it's clear why the application didn't continue. Thoughts?

Reply:
Ok, please add Error log inside if statement, so it will be clear why device stopped

Reply:
I added logging and a unit test to cover the change